src/main/kotlin/com/shortener/service/UrlServiceImpl.kt

```diff
@@ -24,7 +25,8 @@ private val baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build
 class UrlServiceImpl(
     private val urlEntryRepository: UrlEntryRepository,
     private val idEncoder: IdEncoder,
-    private val urlValidator: UrlValidator
+    private val urlValidator: UrlValidator,
+    private val urlEntryRepositoryCacheImpl: UrlEntryRepositoryBase
```

Comment:
architectural question: why does `UrlServiceImpl` care if the repository uses cache or not? It is reflected in the name of the variable.

What bothers me more though, is that now we ahve 2 repositories: one with cache, another without. during autowiring someone might ask for the wrong repository.

Reply:
Probably the implementation is not so coherent as I would have wanted to be. The intention here was to have a simple repo and a caching decorator around it(having the same common interface, `UrlEntryRepositoryBase`), allowing the clients to use either of the two. Why there are still two repos ? Some methods return data which I didn't want it cached. Probably I should've use some kind of factory to generate the correct repo for the client